core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -3801,6 +3803,107 @@ private ExpressionNode rewriteAggregate(ExpressionNode agg, QueryModel model) {
         return agg;
     }
 
+    /**
+     Rewrites
+     SELECT LAST(timestamp) FROM table_name;
+     AND
+     SELECT max(timestamp) FROM table_name;
+     into query which can is optimised search
+     SELECT timestamp from (SELECT timestamp from table_name order by timestamp) limit 1
+     This enables FrameBackwardScan for aggregation queries containing LAST keyword
+     AND
+     Rewrites
+     SELECT FIRST(timestamp) FROM table_name;
+     AND
+     SELECT min(timestamp) FROM table_name;
+     into query which can is optimised search
+     SELECT timestamp from (SELECT timestamp from table_name) limit 1
+     This disables invoking group by workers
+     */
+    private void optimiseAggregateFunctions(QueryModel parent){
+
+        //base condition to stop recursion
+        if(parent== null || parent.getNestedModel() == null)
+          return;
+
+        ObjList<QueryColumn> queryColumns = parent.getBottomUpColumns();
+        QueryModel nestedModel = parent.getNestedModel();
+        CharSequence designatedTimestampColumn = null;
+
+        /**if FIRST/LAST/min/max(column) does not contain designated-timestamp column
+         * OR
+         * another column present in select clause
+         * then don't apply optimisations to base model
+         */
+        if (nestedModel.getTimestamp() == null || queryColumns.size() > 1) {
+            optimiseAggregateFunctions(nestedModel);
+            return;
+        }
+
+        //get designated timestamp column name
+        designatedTimestampColumn =  nestedModel.getTimestamp().token;
+        QueryColumn column = queryColumns.get(0);
+        ExpressionNode ast = column.getAst();
+        CharSequence token= null;
+        CharSequence rhs= null;
+        if(ast != null) {
+            token = ast.token;
+             rhs = ast.rhs == null ? null: ast.rhs.token;
+        }
+
+        /**
+         Type 1 Optimisations: Will be optimised by adding order by clause and changing model type
+         Type 2 Optimisations: Will be done only by changing model type to prevent invoking group by workers
+         */
+        int optimisationType = 0;
+        if(AGGREGATE_FUNCTIONS_OPTIMISED_BY_BACKWARD_SCAN.contains(token.toString()))
```

Comment:
Please avoid redundant object allocations (`token.toString()`). We have non-allocating methods for `CharSequence`s in `Chars`. Same applies to similar calls below.